source/fundamentals/connection/tls.txt

```diff
@@ -0,0 +1,222 @@
+revoked before it connects. You can enable revocation checking using either 
+``MongoClientSettings`` or the connection string.
+
+.. tabs::
+
+   .. tab:: MongoClientSettings
+      :tabid: mongoclientsettings
+
+      To enable revocation checking using ``MongoClientSettings``, set 
+      ``SslSettings.CheckCertificateRevocation`` to ``true``:
+
+      .. code-block:: csharp
+         :emphasize-lines: 5
+
+         var settings = new MongoClientSettings
+         {
+            SslSettings = new SslSettings
+            {
+               CheckCertificateRevocation = true
+            },
+            UseTls = true
+         };
+
+   .. tab:: Connection String
+      :tabid: connectionstring
+
+      To enable revocation checking using a connection string, 
+      assign the connection string parameter ``tlsDisableCertificateRevocationCheck`` 
+      a value of ``false``:
+
+      .. code-block:: csharp
+
+         var mongoClient = new MongoClient("mongodb://user1:password1@sample.host:27017?tls=true&tlsDisableCertificateRevocationCheck=false");
+
+.. note::
+
+   The {+driver-short+} doesn't check revocation by default because this is the default
+   behavior of the ``SslStream`` class in both the 
+   `{+framework+} <https://learn.microsoft.com/en-us/dotnet/api/system.net.security.sslstream.authenticateasclient?view=netframework-{+framework-version+}#System_Net_Security_SslStream_AuthenticateAsClient_System_String_>`__ 
+   and the `.NET standard. <https://learn.microsoft.com/en-us/dotnet/api/system.net.security.sslstream.authenticateasclient?view=netstandard-{+standard-version+}#System_Net_Security_SslStream_AuthenticateAsClient_System_String_>`__
+
+Revocation Checking by Operating System
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The {+driver-short+} supports the following revocation-checking mechanisms differently on 
+Windows, macOS, and Linux:
+
+- :wikipedia:`Online Certificate Status Protocol (OCSP) <Online_Certificate_Status_Protocol>`,
+  a common mechanism for checking revocation 
+- :wikipedia:`OCSP stapling <OCSP_stapling>`, a mechanism in which the server 
+  includes a time-stamped OCSP response to the client along with the certificate
+- :wikipedia:`Certificate revocation lists (CRLs), <Certificate_revocation_list>`,
+  an alternative to OCSP
+
+Windows
+```````
+
+On Windows, the {+driver-short+} supports OCSP, OCSP stapling, and CRLs without OCSP,
+in both the .NET Framework and .NET Core. 
+
+.. warning:: 
+   
+   On Windows, the {+driver-short+} will report a "hard fail" and cancel the TLS 
+   handshake if the OCSP responder is unavailable. Other operating systems and drivers 
+   will report a "soft fail" and continue connecting.
+
+macOS
+`````
+
+On macOS, the {+driver-short+} supports OCSP and OCSP stapling. 
+
+Beginning with .NET Core 2.0, the driver does **not** support CRLs without OCSP.
+
+Linux
+`````
+
+On Linux, the {+driver-short+} supports OCSP, OCSP stapling, and CRLs without OCSP.
+
+
```

Comment:
Can we add a API Documentation section with the methods used on this page?

Reply:
There aren't any methods, but I added a link to `MongoClientSettings`

Reply:
Ah I misread and was thinking the X509Certificate2 object used at the beginning was a method. This works!